woocommerce/woocommerce-android
# [Shipping Labels Revamp] Update Carrier and Saved packages data for better state control

Summary
==========
Partially fix issue #12747 by introducing a series of adjustments to the Carrier and Saved packages data to allow a complete state control with loading and error handling.

The upcoming PR will be focused on triggering the selected package to the main Shipping Labels UI.

Screenshots
==========
https://github.com/user-attachments/assets/3e50c713-66ca-468b-b2d5-cba735535704

How to Test
==========
1. Go to the Order Details of an Order eligible for Shipping Label Creation
2. Hit the Create Shipping Labels button
3. Once inside the new Shipping Labels form, hit the `Select a Package` button
4. Verify that the Carrier and Saved tabs display their loading feedback properly
5. Verify that the `Add Package` button is only enabled for the tab with an actual selected Package
6. Verify that the Custom Package form logic is working as expected

Update release notes:


## Reviewer (or Author, in the case of optional code reviews):

Please make sure these conditions are met before approving the PR, or request changes if the PR needs improvement:
